remote_attestation/rust/src/crypto.rs

```diff
@@ -258,19 +260,88 @@ impl KeyNegotiator {
     }
 }
 
-/// Defines the type of encryptor created by [`KeyNegotiator::create_encryptor`].
-pub enum EncryptorType {
-    /// Defines a server encryptor, which uses server session key for encryption and client session
-    /// key for decryption.
+/// Defines the type of key negotiator and the set of session keys created by it.
+#[derive(Clone)]
+pub enum KeyNegotiatorType {
+    /// Defines a key negotiator which provides server session key for encryption and client
+    /// session key for decryption.
     Server,
-    /// Defines a client encryptor, which uses client session key for encryption and server session
-    /// key for decryption.
+    /// Defines a key negotiator which provides client session key for encryption and server
+    /// session key for decryption.
     Client,
 }
 
+pub struct Signer {
+    /// Parsed PKCS#8 v2 key pair representation.
+    key_pair: EcdsaKeyPair,
+}
+
+impl Signer {
+    pub fn create() -> anyhow::Result<Self> {
+        let rng = ring::rand::SystemRandom::new();
+        let key_pair_pkcs8 =
+            EcdsaKeyPair::generate_pkcs8(&ring::signature::ECDSA_P256_SHA256_FIXED_SIGNING, &rng)
+                .map_err(|error| anyhow!("Couldn't generate PKCS#8 key pair: {:?}", error))?;
+        let key_pair = EcdsaKeyPair::from_pkcs8(
+            &ring::signature::ECDSA_P256_SHA256_FIXED_SIGNING,
+            key_pair_pkcs8.as_ref(),
+        )
+        .map_err(|error| anyhow!("Couldn't parse generated key pair: {:?}", error))?;
+
+        Ok(Self { key_pair })
+    }
+
+    pub fn public_key(&self) -> Vec<u8> {
+        self.key_pair.public_key().as_ref().to_vec()
+    }
+
+    pub fn sign(&self, input: &[u8]) -> anyhow::Result<Vec<u8>> {
+        let rng = ring::rand::SystemRandom::new();
+        let signature = self
+            .key_pair
+            .sign(&rng, input)
+            .map_err(|error| anyhow!("Couldn't sign input: {:?}", error))?
+            .as_ref()
+            .to_vec();
+        Ok(signature)
+    }
+}
+
+pub struct SignatureVerifier {
```

Comment:
If this should be kept in sync with the java version it might be good to add a comment about it here.